src/experiments/74.particles-emitter.tsx

```diff
@@ -0,0 +1,192 @@
+import { useControls } from 'leva'
+import React, { useEffect, useMemo, useRef } from 'react'
+
+import { HTMLLayout } from '~/components/layout/html-layout'
+
+function hexToRgb(hex: string) {
+  const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex)
+  if (!result || !result[1] || !result[2] || !result[3]) return null
+  return `${parseInt(result[1], 16)}, ${parseInt(result[2], 16)}, ${parseInt(
+    result[3],
+    16
+  )}`
+}
+
+interface Particle {
+  x: number
+  y: number
+  size: number
+  speedX: number
+  speedY: number
+  color: string
+  opacity: number
+}
+
+function ParticlesEmitter() {
+  const containerRef = useRef<HTMLDivElement>(null)
+  const mousePosition = useRef({ x: 0, y: 0 })
+  const particles = useRef<Particle[]>([])
+  const animationRef = useRef<number>()
+
+  /* Controls */
+  const {
+    particleCreationRate,
+    particleSize,
+    particleColor,
+    gravityStrength,
+    particleLifespan,
+    emissionRadius,
+    maxParticles,
+    updateFrequency
+  } = useControls({
+    particleCreationRate: { value: 1, min: 0.1, max: 1, step: 0.1 },
+    particleSize: { value: 4, min: 1, max: 10, step: 1 },
+    particleColor: '#FF4D00',
+    gravityStrength: { value: 0.2, min: -1, max: 1, step: 0.01 },
+    particleLifespan: { value: 40, min: 10, max: 100, step: 10 },
```

Comment:
remove the `particle` part from the ones that start with particle to ensure better legibility of the leva props